java-checks/src/main/java/org/sonar/java/checks/StaticFieldInitializationCheck.java

```diff
@@ -36,6 +37,9 @@
 public class StaticFieldInitializationCheck extends AbstractInSynchronizeChecker {
 
   private Deque<Boolean> withinStaticInitializer = new LinkedList<>();
+  private Deque<Boolean> methodUsesLocks = new LinkedList<>();
+  private MethodMatcher lockMethod = MethodMatcher.create().typeDefinition("java.util.concurrent.locks.Lock").name("lock").withoutParameter();
```

Comment:
I would use a `private static final MethodMatcherCollection LOCK_METHODS` instead of declaring two distinct matchers.